account_maturity_date_default/models/account_move.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get("date"):
+            self.mapped("line_ids").filtered(
+                lambda x: (
+                    not x.date_maturity
+                    and x.account_internal_type in {"receivable", "payable"}
+                )
+            ).write({"date_maturity": vals["date"]})
```

Comment:
As a note, this might conflict with the `account_invoice_date_due` module: https://github.com/OCA/account-invoicing/blob/13.0/account_invoice_date_due/models/account_move.py#L30

Can that be a problem?

Reply:
Good point, although I think it doesn't affect, as we already have a maturity date if coming from an invoice, and the other code is changing such date.

Reply:
Yes, both of them are constrained to specific situations, so I guess it is fine...